src/main/java/com/codebutler/android_websockets/WebSocketClient.java

```diff
@@ -72,6 +84,8 @@ public void run() {
                 try {
                     String secret = createSecret();
 
+                    mURI = targetUri;
```

Comment:
Why not move this assignment out of the runnable and into the connect(URI) method, before the Thread is created.  Might be less error prone if we set the URI sooner rather than later.